### PR TITLE
Small Code Updates to support Python3.11

### PR DESCRIPTION
**Pull request by mspi92**
When using cimpyorm with python 3.11 there is an Import Error for `from collections import ..., Iterable, ...`
see also:
```
The Iterable abstract class was removed from collections in Python 3.10. See the deprecation note in the 3.9 [collections docs](https://docs.python.org/3.9/library/collections.html). In the section [Removed](https://docs.python.org/3/whatsnew/3.10.html#removed) of the 3.10 docs, the item

Remove deprecated aliases to [Collections Abstract Base Classes](https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes) from the collections module. (Contributed by Victor Stinner in bpo-37324.)
```

This PR fixes the Code Issue. 

Additionally i adjusted the pylintrc config (got some warnings) and GitlabCI + GithubAction config to support for a broader range of Python Versions being tested

### CODE REVIEW

File: cimpyorm/Model/Schema.py [hunk 1]
```diff
@@ -11,8 +11,8 @@
 import json
 from argparse import Namespace
 import os
-from collections import ChainMap, Iterable, defaultdict
-
+from collections import ChainMap, defaultdict
+from typing import Iterable
```
Comment (TOffergeld) [inline cimpyorm/Model/Schema.py]: Looks like `typing.Iterable` was actually deprecated in favour of `collections.abc.Iterable` in 3.9? See https://docs.python.org/3.9/library/typing.html?highlight=iterable#typing.Iterable

Could just drop support for 3.8 (eol in 2024) maybe?

File: setup.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
     name='cimpyorm',  # Required
     # https://www.python.org/dev/peps/pep-0440/
     # https://packaging.python.org/en/latest/single_source_version.html
-    version='0.9.4',  # Required
+    version='0.9.5',  # Required
```
Comment (TOffergeld) [inline setup.py]:
```suggestion
    version='0.10.0',  # Required
```

If we drop support for 3.7 and potentially 3.8.

**Review by TOffergeld — COMMENTED** [review]
Hi @mspi92, interestingly I still get pinged here but I don't have write access anymore 😄 Let me figure out if someone at IAEW wants to handle this going forward.

Thanks a lot for the PR, I've left two suggestions :)

File: cimpyorm/Model/Schema.py (hunk 1 above)
Comment (mspi92) [reply cimpyorm/Model/Schema.py]: Adjusted it accordingly. Good catch, i did that the other way round and went for the "typing" import. But yes, trend shows that the typing library is becoming more and more deprecated

File: setup.py (hunk 2 above)
Comment (mspi92) [reply setup.py]: Sounds like a good idea. I removed 3.8 from the CI-configs and set the version to 0.10.0

**Comment by mspi92** [comment]
Jo @TOffergeld,
Thanks for the quick review, I guess once you are Maintainer of a project and didn't turn off notifications you'll get notified :D 
Do you know of the person responsible for this repo or was there any agreement on whoever is the maintainer?
Otherwise we can also try to get into contact with the IAEW

**Pull request merged**